modules/vre/spec/jobs/vre_submit1900_job_spec.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe VRE::VRESubmit1900Job do
+  let(:user_struct) do
+    OpenStruct.new(
+      edipi: '1007697216',
+      birls_id: '796043735',
+      participant_id: '600061742',
+      pid: '600061742',
+      birth_date: '1986-05-06T00:00:00+00:00'.to_date,
+      ssn: '796043735',
+      vet360_id: '1781151',
+      loa3?: true,
+      icn: '1013032368V065534',
+      uuid: 'b2fab2b5-6af0-45e1-a9e2-394347af91ef',
+      va_profile_email: 'test@test.com'
+    )
+  end
+  let(:encrypted_user) { KmsEncrypted::Box.new.encrypt(user_struct.to_h.to_json) }
+  let(:user) { OpenStruct.new(JSON.parse(KmsEncrypted::Box.new.decrypt(encrypted_user))) }
+  let(:claim) { create(:veteran_readiness_employment_claim) }
+
+  let(:monitor) { double('VRE::VREMonitor') }
+  let(:exhaustion_msg) do
+    { 'args' => [], 'class' => 'VRE::VRESubmit1900Job', 'error_message' => 'An error occured',
+      'queue' => nil }
+  end
+
+  describe '#perform' do
+    subject { described_class.new.perform(claim.id, encrypted_user) }
+
+    before do
+      allow(VRE::VREVeteranReadinessEmploymentClaim).to receive(:find).and_return(claim)
+    end
+
+    after do
+      subject
+    end
+
+    it 'calls claim.add_claimant_info' do
+      allow(claim).to receive(:send_to_lighthouse!)
+      allow(claim).to receive(:send_to_res)
+
+      expect(claim).to receive(:add_claimant_info).with(user)
+    end
+
+    it 'calls claim.send_to_vre' do
+      expect(claim).to receive(:send_to_vre).with(user)
+    end
+  end
+
+  describe 'raises an exception with email flipper on' do
+    before do
+      allow(VRE::VREVeteranReadinessEmploymentClaim).to receive(:find).and_return(claim)
+      allow(VRE::VREMonitor).to receive(:new).and_return(monitor)
+      allow(monitor).to receive :track_submission_exhaustion
+      Flipper.enable(:vre_trigger_action_needed_email)
+    end
+
+    it 'when queue is exhausted' do
```

Comment:
This "it" seems more like a describe. Maybe this could be restructured to something like:
```
describe "queue exhaustion"
  context "when user has email"
    it "raises an exception and sends an email to the user"
  context "when user has no email"
    it "raises an exception but sends no email"
```
But I think for that to be accurate, you'd also have to add a test like `expect(VANotify::EmailJob).not_to receive(:perform_async)`.